From: Tejun Heo <teheo@suse.de>
Subject: libata: SRST can't be trusted on PMP sil3726
References: 393456

As in sil4726, SRST can't be trusted on sil3726 causing detection
problems under certain configuraitons.  I thought it was from the
Config Disk device but apparently not.

Signed-off-by: Tejun Heo <teheo@suse.de>
---
 drivers/ata/libata-pmp.c |    7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

Index: linux-2.6.25/drivers/ata/libata-pmp.c
===================================================================
--- linux-2.6.25.orig/drivers/ata/libata-pmp.c
+++ linux-2.6.25/drivers/ata/libata-pmp.c
@@ -452,9 +452,12 @@ static void sata_pmp_quirks(struct ata_p
 			/* SError.N need a kick in the ass to get working */
 			link->flags |= ATA_LFLAG_HRST_TO_RESUME;
 
-			/* class code report is unreliable */
+			/* Class code report is unreliable and SRST
+			 * times out under certain configurations.
+			 */
 			if (link->pmp < 5)
-				link->flags |= ATA_LFLAG_ASSUME_ATA;
+				link->flags |= ATA_LFLAG_NO_SRST |
+					       ATA_LFLAG_ASSUME_ATA;
 
 			/* port 5 is for SEMB device and it doesn't like SRST */
 			if (link->pmp == 5)
